### PR TITLE
Add @Uninterruptible annotations to some methods that are called uninterruptible transitively

### PR DESCRIPTION
**Pull request by rkennke**
Add @Uninterruptible annotations to some methods that are called uninterruptible transitively.

### CODE REVIEW

File: substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/traceid/JfrTraceIdMap.java [hunk 1]
```diff
@@ -66,6 +66,7 @@ long getId(int index) {
         return traceIDs[index];
     }
 
+    @Uninterruptible(reason = "Called from uninterruptible code.", mayBeInlined = true)
```
Comment (jiekang) [inline substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/traceid/JfrTraceIdMap.java:69]: The call chain to this goes through `SubstrateJVM.getClassId`. This is currently called by two methods, `Target_jdk_jfr_internal_JVM.getClassIdNonIntrinsic` and `JfrNativeEventWriter.putClass`. The second one is marked uninterruptible. Could you also mark `SubstrateJVM.getClassId` as uninterruptible + mayBeInlined = true in this PR as well? It'll prevent a future bug when native events through `JfrNativeEventWriter` try to `putClass`.

**Comment by jiekang** [comment]
The test code compiles for me now. I have one request to mark an additional method as uninterruptible and will approve after.

File: substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/traceid/JfrTraceIdMap.java (hunk 1 above)
Comment (rkennke) [reply substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/traceid/JfrTraceIdMap.java:69]: Done. I needed to add some more annotations to satisfy the compiler. ;-)

Comment (jiekang) [reply substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/traceid/JfrTraceIdMap.java:69]: So many uninterruptible methods D: heh. Thanks!

**Review by jiekang — APPROVED** [review]

**Review by christianhaeubl — APPROVED** [review]

**Pull request merged**